FhirToDataLake/deploy/templates/ContainerApp/DeployFhirPipelineToContainerApp.json

```diff
@@ -1,420 +1,447 @@
+          "defaultValue": "Standard_LRS",
+          "allowedValues": [
+              "Standard_LRS",
+              "Standard_GRS",
+              "Standard_RAGRS"
+          ],
+          "metadata": {
+              "description": "Storage Account type"
           }
-        }
       }
     },
-    {
-        "type": "Microsoft.Storage/storageAccounts/providers/roleAssignments",
-        "apiVersion": "2018-09-01-preview",
-        "name": "[concat(variables('storageAccountName'), '/Microsoft.Authorization/', guid(uniqueString(variables('storageAccountName'), variables('pipelineName'), 'blob')))]",
-        "dependsOn": [
-            "[resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName'))]",
-            "[resourceId('Microsoft.App/containerApps', variables('pipelineName'))]"
-        ],
+    "variables": {
+      "pipelineContainerName": "fhir-to-datalake",
+      "pipelineName": "[toLower(parameters('pipelineName'))]",
+      "analyticsWorkspaceName": "[concat(variables('pipelineName'), '-LogAnalytics')]",
+      "environmentName": "[if(empty(parameters('containerAppEnvironmentName')), concat(variables('pipelineName'), '-Environment') ,parameters('containerAppEnvironmentName'))]",
+      "jobInfoQueueName": "[concat(variables('pipelineName'), 'jobinfoqueue')]",
+      "jobInfoTableName": "[concat(variables('pipelineName'), 'jobinfotable')]",
+      "metadataTableName": "[concat(variables('pipelineName'), 'metadatatable')]",
+      "resourceTags": {"FhirAnalyticsPipeline": "FhirToDataLake", "FhirSchemaVersion": "v0.6.0"},
+      "storageAccountName": "[concat(substring(replace(variables('pipelineName'), '-', ''), 0, min(11, length(replace(variables('pipelineName'), '-', '')))), uniquestring(resourceGroup().id, variables('pipelineName')))]",
+      "storageBlobDataContributerRoleId": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Authorization/roleDefinitions/', 'ba92f5b4-2d11-453d-a403-e96b0029c9fe')]",
+      "storageQueueDataContributerRoleId": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Authorization/roleDefinitions/', '974c5e8b-45b9-4653-ba55-5f855dd0fb88')]",
+      "storageTableDataContributerRoleId": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Authorization/roleDefinitions/', '0a9a7e1f-b9d0-4cc4-a60d-0319b160aaa3')]",
+      "filterConfigImageReference": "[toLower(parameters('filterConfigImageReference'))]",
+      "customizedSchemaImageReference": "[toLower(parameters('customizedSchemaImageReference'))]",
+      "schedulerCronExpression": "[if(equals(parameters('schedulerIntervalInMinutes'), 5), '0 */5 * * * *', if(equals(parameters('schedulerIntervalInMinutes'), 30), '0 */30 * * * *', if(equals(parameters('schedulerIntervalInMinutes'), 60), '0 0 * * * *', if(equals(parameters('schedulerIntervalInMinutes'), 1440), '0 0 0 * * *', parameters('schedulerCronExpression')))))]"
```

Comment:
Seems **schedulerCronExpressio** will be empty if **parameters('schedulerIntervalInMinutes')** is 0 and **parameters('schedulerCronExpression')** is empty.

Reply:
yes, it will be empty. this is an invalid case, if the customer sets `schedulerIntervalInMinutes` to 0, then he needs to set the `schedulerCronExpression`, otherwise the pipeline will throw ConfigurationErrorException at the configuration validator, and the provision will fail